move all apt repos to keyrings

rrotter opened the pull request:
All non-OS keys belong in `/etc/apt/keyrings` and each repo should be individually pinned to a key. The `puppet/apt` module detects that you are using a keyring based on the use of the `'key'=>'name'` field, which isn't needed or allowed for classic keys since they all get cat'ed into the same file.

`/etc/apt/keyrings` removes the need to convert between binary and "ascii armored" keys: the format is specified by the file extension (.gpg/.asc).

The file names are arbitrary (they only need to be unique, and `puppet/apt` deals with both writing and linking them), but I tried to be somewhat consistent with the naming:
* Use the domain basename if it's clear what they are hosting:
  * adoptium.com => adoptium.gpg
* If the domain isn't com/net/org/edu, include it:
  * elastic.co => elastic.co.asc
* Prepend the name of the source if the domain is opaque (and keep the tld because: what's a "sury"??):
  * php-community + https://packages.sury.org/php/ => php-community-sury.org.gpg
* Binary files are .gpg, ascii files are .asc, this is an OS feature.

Note the `local` repo is managed in lens, not here, the corresponding environment has been updated there and does not depend on these changes.

rrotter committed:
remove dead code for purging `apt-get update` cron

rrotter committed:
use keyrings for all apt repos

- every instance of apt::source updated to use key => name field, which
  causes `puppetlabs/apt` module to use a keyring for the repo
- tests updated where applicable (if test was expecting the no longer
  used key => id field)

rrotter commented:
Also, repo for mono is broken. Fixing it is out of scope here, but I added a note about it in the PR (see: manifests/profile/apt/mono.pp), because atm it's not even clear anyone is aware of this.

rrotter committed:
configure puppet apt on Ubuntu

Puppet apt repo was managed by puppet on Debian but not Ubuntu. Putting
it under puppet management eliminates the last place where 3rd party
keys in trusted.gpg or trusted.gpg.d were needed.

rrotter committed:
rm trusted.gpg, trusted.gpg.d/puppet*.gpg

- trusted.gpg was only used for 3rd party repos, now moved to keyrings
- trusted.gpg.d is used for 3rd party and OS repos. All 3rd party repos
  should be removed, but
  - puppet is the most common repo written here (w/ many names over the
    years, hence the pattern match)
  - VM provisioning will still add the puppet repo, so it may need to be
    removed by puppet to keep it gone. Other repo keys can be manually
    deleted and will not be created by any current automation.

rrotter committed:
delete redundant sources.list.d entries on Ubuntu

rrotter commented:
Had to add cleanup of sources.list.d for Ubuntu because we ended up w/ multiple conflicting puppet apt configs. We should probably manage all the apt files for Ubuntu like we do for Debian, but that's out of scope.

skorner reviewed (approved):
This all makes sense and looks good to me. I appreciate the thought given to the naming conventions and listing them here, as well as noting issues you came across that are out of scope but could use attention at some point.

rrotter committed:
mirror adoptium key in /files

Source server does not set Last-Modified so puppet always assumes the
file is new and reinstalls it on every run.

See:
- `curl -I`
- https://www.puppet.com/docs/puppet/7/types/file.html#file-attribute-source
